### ISSUE TITLE
Use favicon for RSS feeds

### ISSUE THREAD

**Issue by schiessle** [friendica/friendica#7695]
I really like to use Friendica to subscribe to RSS feeds. This way I have everything in one stream and can easily re-share interesting articles. But what really disturbs me is the ugly icon for RSS feeds:

![image](https://user-images.githubusercontent.com/1589737/66115640-88597380-e5d1-11e9-8881-aad556917e82.png)

I think it would look way nicer if Friendica would fetch the favicon and use it as "avatar" for the stream. If no Favicon exists a RSS logo could be used as a placeholder so that it is easier to distinguish RSS feeds from other contacts without a avatar.

Maybe it would be a nice additional option if the user could upload their own avatar for their RSS feeds?

**Comment by MrPetovan** [comment]
That's a great idea, it's always been bugging me. What about favicon that aren't big enough to fill the profile image size?

**Comment by MrPetovan** [comment]
Also you can already change the image of any of your contact. Go to your RSS contact page, click on "Advanced", And fill the "New photo from URL" field.

**Comment by schiessle** [comment]
> Also you can already change the image of any of your contact. Go to your RSS contact page, click on "Advanced", And fill the "New photo from URL" field.

This solves the issue for me! :slightly_smiling_face: I keep the issue open in case someone likes the idea of fetching a default image somehow automatically (favicon, etc). But I'm also happy if we just close it. I will leave the decision to the developers

**Comment by AlfredSK** [comment]
The thing with changing the avatar in the contact settings is that the local avatar does not federate. When you reshare a rss post the avatar is lost on transport to remote servers.

**Comment by annando** [comment]
I can easily change the probing mechanism to have a look for a favicon as well. This would also mean that on reshared posts the other side would fetch the avatar from the favicon as well.

@schiessle How are the exact rules for the favicon? Do you have some short documentation? AFAIK the file had to be named "favicon.ico", but I'm not totally sure if there had been some changes in the specification over the years.

**Comment by MrPetovan** [comment]
> @schiessle How are the exact rules for the favicon? Do you have some short documentation? AFAIK the file had to be named "favicon.ico", but I'm not totally sure if there had been some changes in the specification over the years.

First, you can look for the `manifest.json`:
```
{
  "short_name": "Maps",
  "name": "Google Maps",
  "icons": [
    {
      "src": "/images/icons-192.png",
      "type": "image/png",
      "sizes": "192x192"
    },
    {
      "src": "/images/icons-512.png",
      "type": "image/png",
      "sizes": "512x512"
    }
  ],
  "start_url": "/maps/?source=pwa",
  "background_color": "#3367D6",
  "display": "standalone",
  "scope": "/maps/",
  "theme_color": "#3367D6"
}
```
Then if it isn't present, you have to scan the page tags for these potential values:
```html
// More likely to be the right size for profile picture
<link rel="apple-touch-icon" href="images/friendica-128.png"/>

// Modern
<link rel="icon" href="/path/to/favicon-228.png" sizes="228x228">
...
<link rel="icon" href="/path/to/favicon-228.png" sizes="16x16">

// Legacy
<link rel="shortcut icon" href="favicon.ico" />
<link rel="icon" href="favicon.ico" />
```
If all else fails you can try for a `favicon.ico` file.

If you have to retrieve an `.ico` file, it can contain multiple sizes, not just the classic 16x16 pixels image. Ideally, we should extract the largest image in the set.

**Comment by hoergen** [comment]
Another thing one could consider is the Open Graph Protocol? https://ogp.me/ 
If everything else fails. 

Just a thought .. when a contact should be established, you could give the user the option to select out of your "findings" (different sizes of favicons or ogp) which graphic you want to use for that contact ?

**Comment by MrPetovan** [comment]
> Just a thought .. when a contact should be established, you could give the user the option to select out of your "findings" (different sizes of favicons or ogp) which graphic you want to use for that contact ?

Yes, but it doesn't make sense for regular contacts belonging to people who set their profile picture. So it would be unused most of the time because only one picture would be shown for most contacts. Only RSS feed are concerned by this.

**Comment by hoergen** [comment]
As far as I tested it an image is fetched, when adding a RSS feed. Can this issue be closed?

**Comment by annando** [comment]
We don't fetch a favicon. We fetch a provided image. But images aren't provided at any time. So this is something I think it would be nice.
